Homework_7/src/ParserPackage/FileParser.java

```diff
@@ -0,0 +1,108 @@
+package ParserPackage;
+
+import java.io.*;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class FileParser {
+    private File fileForReading ;
+    public File containsFile(String fileName){
+        File bookFile = new File("src\\"+fileName);
+        if (!bookFile.exists()) {
+           throw new RuntimeException("Book is not found !");
+        }
+        fileForReading=bookFile;
+        return bookFile;
+    }
+    public  List<String> clearningStrings (List<String> inputStringSequence){
+        Pattern regularExpression =Pattern.compile("((^\\W*)([a-zA-Z]+\\'?\\-?[a-zA-Z]*)(\\W*$))");
+        return inputStringSequence.stream()
+               // .map(String::toLowerCase)
+                .map(s->{
+                    Matcher matcher =regularExpression.matcher(s);
+                    return matcher.matches() ? matcher.group(3) :"";
+                })
+                .filter(s->!s.isEmpty())
+                .collect(Collectors.toList());
+    }
+    public List<String> getStringsFromFile(){
+        ArrayList<ArrayList<String>> internalStorage=new ArrayList<>();
+        try ( BufferedReader fileStringReader = new BufferedReader(new FileReader(fileForReading))){
+            String line;
+            while ((line=fileStringReader.readLine())!=null) {
+                ArrayList<String> nestedStringCollection = new ArrayList<>();
```

Comment:
from var type I see that it Collection, you don't need to use word collection in var name